plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java

```diff
@@ -388,19 +403,37 @@ protected String addDBMetadataToRequest(
         return additionalMetadata.toString();
     }
 
+    private boolean isRequiresFullyQualifiedName(@NotNull DBSObject object, @Nullable DBCExecutionContext context) {
+        if (context == null || context.getContextDefaults() == null
+        ) {
+            return true;
+        }
+        DBSObject parent = object.getParentObject();
+        return parent != null && !(
+            parent.equals(context.getContextDefaults().getDefaultCatalog())
+                || parent.equals(context.getContextDefaults().getDefaultSchema()));
+    }
```

Comment:
```suggestion
    private boolean isRequiresFullyQualifiedName(@NotNull DBSObject object, @Nullable DBCExecutionContext context) {
        if (context == null || context.getContextDefaults() == null) {
            return true;
        }
        DBSObject parent = object.getParentObject();
        DBSObject defaults= context.getContextDefaults();
        return parent != null && !(parent.equals(defaults.getDefaultCatalog()) || parent.equals(defaults.getDefaultSchema()));
    }
```

Reply:
By the way, `context.getContextDefaults()` can be nullable

Reply:
context.getContextDefaults() == null?